scripts/Calibration/tofpd/diagnostics.py

```diff
@@ -264,10 +264,11 @@ def difc_plot2d(calib_new, calib_old=None, instr_ws=None, mask=None, vrange=(0,
             value_array.append(percent)
 
     # Use the largest solid angle for circle radius
-    sample_position = info.samplePosition()
     maximum_solid_angle = 0.0
+    LoadEmptyInstrument(InstrumentName=instr_name, OutputWorkspace=f"{instr_name}_instr")
+    det_tmp = SolidAngle(InputWorkspace=f"{instr_name}_instr", OutputWorkspace="detector_tmp")
     for det_id in range(info.size()):
-        maximum_solid_angle = max(maximum_solid_angle, delta.getDetector(det_id).solidAngle(sample_position))
+        maximum_solid_angle = max(maximum_solid_angle, det_tmp.readY(det_id)[0])
```

Comment:
The variable `det_id` here is actually a workspace index. It would add clarity to rename the variable.